specifyweb/frontend/js_src/lib/components/QueryBuilder/Wrapped.tsx

```diff
@@ -121,12 +121,18 @@ export function QueryBuilder({
     [queryResource, model, autoRun]
   );
   const [state, dispatch] = React.useReducer(reducer, pendingState);
+
+  const initialFields = React.useRef<string>('');
+
   React.useEffect(() => {
+    const initialState = buildInitialState();
     dispatch({
       type: 'ResetStateAction',
       state: buildInitialState(),
```

Comment:
don't call `buildInitialState()` here again  reuse the `initialState` variable you defined above